### PR TITLE
Python documentation for onnxruntime-training

### PR DESCRIPTION
**Pull request by baijumeswani**
This pull request adds Python API documentation for `onnxruntime-training` for:
- ORTModule
- On device training

Changes currently staged at: https://baijumeswani.github.io/onnxruntime/docs/api/python/index.html

### CODE REVIEW

**Review by snnn — DISMISSED** [review]

**Comment by xadupre** [comment]
It looks like you merged the documentation for onnxruntime and onnxruntime-training. Is it intended? A while ago, I created a PR to extend the documentation, maybe there are some pieces you can reuse: #9246.

**Comment by baijumeswani** [comment]
> It looks like you merged the documentation for onnxruntime and onnxruntime-training. Is it intended?

Yes, documentation for inference and training is intentionally kept together to make it constistent with other API documentation and to showcase that training is really just an extension of inference in ort.

> A while ago, I created a PR to extend the documentation, maybe there are some pieces you can reuse: https://github.com/microsoft/onnxruntime/pull/9246.

Sure. Are there any particular pieces from there that you would like me to port over here? Would you like to address that in a follow up PR since you have done the work in that PR already?

**Comment by xadupre** [comment]
The documentation was about TrainingSession and OrtTrainer. It is only interesting if they remain as part as the public API.

File: docs/python/inference/conf.py [hunk 1]
```diff
@@ -10,8 +10,6 @@
 import shutil  # noqa: F401
```
Comment (xadupre) [inline docs/python/inference/conf.py:10]: If the documentation of onnxruntime and onnxruntime-traning are merged, it woud be better to move this file (and index.rst) the folder python, maybe remove one level of subfolders.

Comment (baijumeswani) [reply docs/python/inference/conf.py:10]: Updated the folder structure.

File: docs/python/inference/index.rst [hunk 2]
```diff
@@ -8,6 +8,22 @@ or the `Github project <https://github.com/microsoft/onnxruntime/>`_.
...
+    :caption: Inference
 
     tutorial
     api_summary
+
+.. toctree::
+   :maxdepth: 1
+   :caption: LARGE MODEL TRAINING
+
+   ortmodule/overview
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply docs/python/inference/index.rst:20]: As we discussed the index.rst file has access to the ortmodule/overview directly.

File: docs/python/learning_on_the_edge/overview.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Overview
+=========
+
+Learning on the edge refers to the process of training a model on an edge device, such as mobile phones, embedded devices, gaming consoles, web browsers, etc. This is in contrast to training a model on a server or a cloud. Training on the edge is useful when the data is sensitive and cannot be shared with a server or a cloud. It is also useful for the task of personalization where the model needs to be trained on the user's device.
+
+`onnxruntime-training` offers an easy to use set of APIs and tools to efficiently train and infer a wide range of ONNX models on edge devices. The training process is divided into two phases:
```
Comment (xadupre) [inline docs/python/learning_on_the_edge/overview.rst:6]: an esay way

File: tools/doc/builddoc.sh [hunk 4]
```diff
@@ -17,12 +17,5 @@ export PYTHONPATH=$3/$4:$PYTHONPATH
...
-# Inference doc
-$1/python -m sphinx -j1 -v -T -b html -d $3/docs/inference/_doctrees/html $2/docs/python/inference $3/docs/inference/html
-$1/python -u $2/tools/doc/rename_folders.py $3/docs/inference/html
-# (cd $3/docs/inference/html && zip -r $3/docs/python_inference_doc.zip .)
-
-# Training doc
-$1/python -m sphinx -j1 -v -T -b html -d $3/docs/training/_doctrees/html $2/docs/python/training $3/docs/training/html
-$1/python -u $2/tools/doc/rename_folders.py $3/docs/training/html
-# (cd $3/docs/training/html && zip -r $3/docs/python_training_doc.zip .)
+$1/python -m sphinx -j1 -v -T -b html -d $3/docs/_doctrees/html $2/docs/python/ $3/docs/html
```
Comment (xadupre) [inline tools/doc/builddoc.sh:20]: -j1 means using one one process to generate the documentation. It should work without it.

**Review by xadupre — DISMISSED** [review]

File: docs/python/learning_on_the_edge/training_artifacts.rst [hunk 5]
```diff
@@ -0,0 +1,141 @@
+Generating the Training Artifacts
```
Comment (natke) [inline docs/python/learning_on_the_edge/training_artifacts.rst:1]: Nit: Generating -> Generate

File: docs/python/learning_on_the_edge/training_api.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
+Training the Model on the Device
```
Comment (natke) [inline docs/python/learning_on_the_edge/training_api.rst:1]: Nit: Training -> Train

File: docs/python/learning_on_the_edge/training_artifacts.rst (hunk 5 above)
Comment (natke) [inline docs/python/learning_on_the_edge/training_artifacts.rst:1]: Would this page be better described as "Prepare for training"?

**Review by natke — APPROVED** [review]
Looks good. A few minor comments. I have not reviewed the API contents in detail though

**Review by natke — APPROVED** [review]

**Review by snnn — APPROVED** [review]

**Pull request merged**